elf/segmentation/watershed.py

```diff
@@ -117,6 +123,8 @@ def distance_transform_watershed(input_, threshold, sigma_seeds,
     else:
         hmap = alpha * input_ + (1. - alpha) * dt
 
+    hmap = hmap.astype('uint8') # change back to uint8, so as to work with Vigra watershedsNew
```

Comment:
Casting to `uint8` here is dangerous. The height map should be `float32`.
If we have the check for `float32` above, we don't need to cast here.

Reply:
My mistake - float32 works too! However, I was having problems with the height map being float64